src/main/java/com/gmail/woosay333/onlinebookstore/controller/OrderController.java

```diff
@@ -0,0 +1,60 @@
+package com.gmail.woosay333.onlinebookstore.controller;
+
+import com.gmail.woosay333.onlinebookstore.dto.order.OrderRequestDto;
+import com.gmail.woosay333.onlinebookstore.dto.order.OrderResponseDto;
+import com.gmail.woosay333.onlinebookstore.dto.status.OrderStatusRequestDto;
+import com.gmail.woosay333.onlinebookstore.entity.User;
+import com.gmail.woosay333.onlinebookstore.service.order.OrderService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+@RequiredArgsConstructor
+@Tag(name = "Order management.",
+        description = "Endpoints for managing orders.")
+public class OrderController {
+    private final OrderService orderService;
+
+    @PostMapping
+    @PreAuthorize("hasAnyRole('ROLE_USER','ROLE_ADMIN')")
+    @Operation(summary = "Place an order.",
+            description = "Place an order by the user.")
+    public void placeOrder(
+            @Valid @RequestBody OrderRequestDto order,
+            @AuthenticationPrincipal User user) {
+        orderService.place(order, user);
+    }
+
+    @GetMapping
+    @PreAuthorize("hasAnyRole('ROLE_USER','ROLE_ADMIN')")
+    @Operation(summary = "Get all orders.",
+            description = "Get all orders of the current user.")
+    public List<OrderResponseDto> getOrders(
+            @AuthenticationPrincipal User user) {
+        return orderService.getAllOrders(user);
+    }
+
+    @PatchMapping("/{id}")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN')")
+    @Operation(summary = "Update an order status.",
+            description = "Update an order status by id")
+    public void updateOrderStatus(
+            @Valid @RequestBody OrderStatusRequestDto statusRequestDto,
+            @PathVariable Long id,
+            @AuthenticationPrincipal User user) {
+        orderService.updateStatus(statusRequestDto, id, user.getId());
```

Comment:
Why sometimes full user object is passed to service method and sometimes it is just id, is it possible to make same in all cases?